src/main/java/org/opentripplanner/middleware/utils/JsonUtils.java

```diff
@@ -90,18 +90,18 @@ private static boolean isResponseOk(HttpResponse<String> response) {
     /**
      * Utility method to parse generic object from HTTP response.
      */
-    public static <T> T getPOJOFromHttpBody(HttpResponse<String> response, Class<T> clazz) {
+    public static <T> T getPOJOFromHttpBody(HttpResponseValues response, Class<T> clazz) {
         return isResponseOk(response)
-            ? getPOJOFromJSON(response.body(), clazz)
+            ? getPOJOFromJSON(response.responseBody, clazz)
```

Comment:
Like right here, this calls `getPOJOFromJSON` which merely calls `mapper.readValue(json, clazz)`. The `mapper` can also take in an InputStream, so technically it'd be possible to read the response body just once using the mapper instead of first reading it into a string and then parsing the string.

Reply:
The response body has to be completely consumed in the response handler. After this the connection will be closed.